akka-http-core/src/main/scala/akka/http/scaladsl/Http.scala

```diff
@@ -86,7 +88,7 @@ class HttpExt private[http] (private val config: Config)(implicit val system: Ex
         case _                 ⇒ httpLayer atop tlsStage
       }
 
-    serverBidiFlow
+    GracefulTerminatorStage(system, settings) atop serverBidiFlow
```

Comment:
this is where we instal the logic

Reply:
Do we have any feeling for how much overhead this feature causes? I guess if it's noticeable we could make it configurable so people for whom unbind+timeout is sufficient can opt out of it?

Reply:
I think not much... it is a "pass through" most of the time, which should be fused

Reply:
And while terminating it should not matter that much since goal is to quickly send 500s anyway

Reply:
I have not yet benchmarked though